### ISSUE TITLE
Return 0 for unseen labels of a Categorical/Nominal Variable in Test Set or in Production

### ISSUE THREAD

**Issue by SurajitTest** [solegalli/feature_engine#86]
The given below 2 questions can be answered with this Feature Request 
Question -1 : In case a categorical variable in the train set does not have a Rare' label based on values in parameters 'tol' and 'n_categories' , how do we handle unseen values in the test set or in production ? 

Question-2 : In the absence of a 'Rare' label in a categorical variable in the train set based on values in parameters 'tol' and 'n_categories', is there any similar mechanism in Feature-Engine like 'handle_unknown='ignore'' as used by sklearn OneHotEncoder for any of the categorical encoding methods implemented in Feature-Engine ?

**Comment by solegalli** [comment]
Hi, thanks for the request! To re-phrase the request:

the one hot encoder from Feature-engine, should have the option to handle labels that have not been seen in the train set, in a similar way to the 'handle_unknown='ignore' from feature engine. That is, for unseen categories, the encoder will return 0 in all the dummy variables.

**Comment by solegalli** [comment]
Now that we can wrap the OneHotEncoder from scikit-learn that already has this functionality, by using the SklearnTransformerWrapper, do we still need to add this functionality to Feature-engine's one hot categorical encoder?

Feels to me that this issue is resolved? @SurajitTest?

**Comment by SurajitTest** [comment]
Hi Sole,
            'Now that we can wrap the OneHotEncoder from scikit-learn that already has this functionality, by using the SklearnTransformerWrapper' - this is very true. But, if the given below is implemented , for unseen labels in the test set, feature engine can stand independently without the use of sklearn(in terms of using the SklearnTransformerWrapper)

the one hot encoder from Feature-engine, should have the option to handle labels that have not been seen in the train set, in a similar way to the 'handle_unknown='ignore' from sklearn. That is, for unseen categories, the encoder will return 0 in all the dummy variables.

**Comment by solegalli** [comment]
Feels to me that using the SklearnTransformerWrapper is a small price to pay for using functionality that already exists and is well supported by the sklearn community.

At the back, you can use the new transformer DropFeatures, to drop the original categorical variables from the dataframe, and this way you will end up with a similar result as that returned by Feature-engine's one hot encoder.
